src/plugins/irc/irc-server.h

```diff
@@ -248,6 +257,8 @@ struct t_irc_server
     time_t last_user_message;       /* time of last user message (anti flood)*/
     time_t last_away_check;         /* time of last away check on server     */
     time_t last_data_purge;         /* time of last purge (some hashtables)  */
+    int authentication_method;      /* authentication method used to login   */
+    int sasl_mechanism_used;        /* SASL method used at login time        */
```

Comment:
Hi,

Sorry for the late check of this PR.

Why adding `sasl_mechanism_used` there? You already have the server option `sasl_mechanism` that contains the mechanism used to authenticate in case SASL was used for authentication.

Is it because reading the server option is not so easy, because of value inheritance from `irc.server_default.sasl_mechanism`?
If that's the case, we could consider instead adding something in API (like an info) that could be used from evaluated string to get the inherited value of any option (including IRC server options).

Something like this, considering in this example that `irc.server.libera.sasl_mechanism` == `null` and `irc.server_default.sasl_mechanism` == `plain`:

    /eval -n ${info:option_value,irc.server.libera.sasl_mechanism}
    == [plain]

Reply:
I think I wanted to create a record of what was actually used, because you can change the option while you are still connected.

Reply:
OK makes sense.